### PR TITLE
Add a basic e2e test for egress network policy creation.

### PR DESCRIPTION
**Pull request by omris94**
### Description

Add a basic e2e test for egress network policy creation.

### Testing

Describe how this can be tested by reviewers. Be specific about anything not tested and reasons why. If this library has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors.

Please include any manual steps for testing end-to-end or functionality not covered by unit/integration tests.

Also include details of the environment this PR was developed in (language/platform/browser version).

- [x] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [x] I have added documentation for new/changed functionality in this PR and in github.com/otterize/docs

### CODE REVIEW

File: .github/workflows/netpol-e2e-test.yaml [hunk 1]
```diff
@@ -89,6 +89,9 @@ jobs:
       - name: Wait for Otterize
         run: |-
           kubectl wait pods -n otterize-system -l app=intents-operator --for condition=Ready --timeout=360s
+          # wait for webhook to be ready
```
Comment (NetanelBollag) [inline .github/workflows/netpol-e2e-test.yaml:92]: Oh yes thank you for that!!!

File: .github/workflows/netpol-e2e-test.yaml [hunk 2]
```diff
@@ -230,10 +237,133 @@ jobs:
...
+          
+          # should not work at first because there is no allow DNS netpol
+          echo check client log
+          wait_for_log $CLI1_POD 10 "curl timed out"
+          
+          # should be blocked (using 3 because the log should repeat itself every 3 lines)
+          echo check client other log
+
+          wait_for_log $CLI2_POD 10 "curl timed out"
+
```
Comment (NetanelBollag) [inline .github/workflows/netpol-e2e-test.yaml:347]: maybe add echo for every step?

Comment (omris94) [reply .github/workflows/netpol-e2e-test.yaml:347]: I'll do it

File: .github/workflows/netpol-e2e-test.yaml [hunk 3]
```diff
@@ -230,10 +237,133 @@ jobs:
...
+          source .github/workflows/test-bashrc.sh
+          
+          kubectl apply -f https://docs.otterize.com/code-examples/automate-network-policies/intents.yaml
+          
+          # should not work at first because there is no allow DNS netpol
+          echo check client log
+          wait_for_log $CLI1_POD 10 "curl timed out"
+          
+          # should be blocked (using 3 because the log should repeat itself every 3 lines)
+          echo check client other log
```
Comment (NetanelBollag) [inline .github/workflows/netpol-e2e-test.yaml:344]: IMHO it's better:
```suggestion
          echo "check client other log"
```

File: .github/workflows/netpol-e2e-test.yaml [hunk 4]
```diff
@@ -230,10 +237,133 @@ jobs:
...
+           # should work because there is an applied intent and allowDNS netpol
+          echo check client log
+          wait_for_log $CLI1_POD 10 "Hi, I am the server, you called, may I help you?"
+          
+           # should be blocked (using 3 because the log should repeat itself every 3 lines)
+          echo check client other log
+
+          wait_for_log $CLI2_POD 10 "curl timed out"
+  
+
```
Comment (NetanelBollag) [inline .github/workflows/netpol-e2e-test.yaml:359]: This test is actually checking nothing is totally broken when egress enabled, but I suggest there will be another service, `server2` that only `client` will call (won't be called by `client-other`). When intents applied we should expect communication with `server2` to stop, and would assert that egress policy actually work with blocking the *client* un-declared calls to servers

**Review by NetanelBollag — COMMENTED** [review]
Generally look good, the only thing is missing IMHO is to actually test egress logic itself

**Pull request merged**